app/src/main/java/com/feedhenry/securenativeandroidtemplate/features/network/presenters/UploadNotesPresenter.java

```diff
@@ -0,0 +1,213 @@
+import com.feedhenry.securenativeandroidtemplate.R;
+import com.feedhenry.securenativeandroidtemplate.domain.Constants;
+import com.feedhenry.securenativeandroidtemplate.domain.callbacks.Callback;
+import com.feedhenry.securenativeandroidtemplate.domain.configurations.ApiServerConfiguration;
+import com.feedhenry.securenativeandroidtemplate.domain.configurations.AppConfiguration;
+import com.feedhenry.securenativeandroidtemplate.domain.models.Note;
+import com.feedhenry.securenativeandroidtemplate.domain.repositories.NoteRepository;
+import com.feedhenry.securenativeandroidtemplate.domain.services.AuthStateService;
+import com.feedhenry.securenativeandroidtemplate.domain.services.NoteCrudlService;
+import com.feedhenry.securenativeandroidtemplate.features.network.views.UploadNotesView;
+import com.feedhenry.securenativeandroidtemplate.mvp.components.HttpHelper;
+import com.feedhenry.securenativeandroidtemplate.mvp.presenters.BasePresenter;
+
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.TimeUnit;
+
+import javax.inject.Inject;
+import javax.net.ssl.SSLSocketFactory;
+import javax.net.ssl.X509TrustManager;
+
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+/**
+ * Created by weili on 27/10/2017.
+ */
+
+public class UploadNotesPresenter extends BasePresenter<UploadNotesView> {
+
+    private static final String TAG = "UploadNotesPresenter";
+    public static final MediaType JSON
+            = MediaType.parse("application/json; charset=utf-8");
+
+    private NoteRepository noteRepository;
+    private AuthStateService authStateService;
+    private ApiServerConfiguration apiServerConfiguration;
+
+    private UploadNotesTask uploadNotesTask;
+
+    @Inject
+    public UploadNotesPresenter(NoteRepository noteRepos, AuthStateService authStateService, AppConfiguration appConfiguration){
+        this.noteRepository = noteRepos;
+        this.authStateService = authStateService;
+        this.apiServerConfiguration = appConfiguration.getAPIServerConfiguration();
+    }
+
+    public void checkClientStatus() {
+        new ClientStatusCheckerTask().execute();
+    }
+
+    public void uploadNotes() {
+        uploadNotesTask = new UploadNotesTask();
+        uploadNotesTask.execute();
+    }
+
+    public void cancelUpload() {
+        if (uploadNotesTask != null) {
+            uploadNotesTask.cancel(true);
+        }
+    }
+
+    private class ClientStatus {
+        private boolean hasPermission;
+        private long numberOfNotes;
+
+        ClientStatus(boolean hasPerm, long numberOfNotes) {
+            this.hasPermission = hasPerm;
+            this.numberOfNotes = numberOfNotes;
+        }
+    }
+
+    private class ClientStatusCheckerTask extends AsyncTask<Void, Void, ClientStatus> {
+
+        private Exception error;
+        String requiredRole = Constants.ACCESS_CONTROL_ROLES.ROLE_MOBILE_USER;
```

Comment:
Maybe we can use the `api-access` role for interactions with the API? - [keycloak/secure-app-realm.json](https://github.com/feedhenry/mobile-security/blob/master/projects/keycloak/secure-app-realm.json#L189-L196). The secure-user, secure-user2 and superuser are in the API Access Keycloak group so those users can interact with the API and upload notes. The API Server note route would need to be protected by this role then too to limit API access to only users in that group with something like: `keycloak.protect('realm:api-access')`

Reply:
yep, good point. will change this